Python/Product/Profiling/Profiling/StandaloneTargetView.cs

```diff
@@ -108,7 +108,7 @@ public StandaloneTargetView(IServiceProvider serviceProvider, StandaloneTarget t
             if (IsValid) {
                 return new StandaloneTarget {
                     PythonInterpreter = CanSpecifyInterpreterPath ? null : Interpreter.GetInterpreter(),
-                    InterpreterPath = CanSpecifyInterpreterPath ? InterpreterPath : null,
+                    InterpreterPath = CanSpecifyInterpreterPath ? null : InterpreterPath,
```

Comment:
It feels like the original code here intended either one or the other property to be null, depending on whether `CanSpecifyInterpreterPath` is set. Is that changing by design?

Reply:
Hi @int19h, the `InterpreterPath` is not used anywhere in the original code, because existing python interpreters just use id to find the configuration. However, the custom interpreter constructs `VisualStudioInterpreterConfiguration` using `runTarget.InterpreterPath`, that's why I swapped them so that actual values can be returned.

Reply:
Looking at the docstrings for the properties involved, I'm still confused about this. `CanSpecifyInterpreterPath` is basically whether the textbox for that is enabled in the UI or not, right? So if it is `true`, shouldn't we be returning the target with `InterpreterPath` set from the property (which in turn gets it from the textbox)? In which case it would also make sense to have `PythonInterpreter` set to null, as it was before the change.

This would then allow `ProfileStandaloneTarget` to distinguish `PythonInterpreter` objects based solely on `Interpreter == null`. In retrospect, I think that was actually the original intent of this code as it was written. Perhaps the real bug here all along was that `PythonInterpreter` wasn't being set correctly in this case?

Reply:
Your comments make sense to me. But I don't think the bug was `PythonInterpreter` wasn't being set correctly. It only has one attribute, which is `id`. `ProfileStandaloneTarget` won't be able to config the custom interpreter with only the `id`, it needs the `InterpreterPath` returned from `GetTarget()`. But I do agree `CanSpecifyInterpreterPath`  is a bit confusing here, the good thing is that it's not breaking anything. Maybe I can just git rid of this check `InterpreterPath = CanSpecifyInterpreterPath ? null : InterpreterPath` here and just return `InterpreterPath` since it's not affecting anything? Thanks again for taking a look Pavel!!

Reply:
I meant that it was improperly set to a `PythonInterpreter` object with null `Id`, rather than to a null `PythonInterpreter` reference. Note that in `ProfileStandaloneTarget`, there's a check for that null, and a different branch is taken in that case: 
```cs
            LaunchConfiguration config;
            if (runTarget.PythonInterpreter != null) {
                var registry = session._serviceProvider.GetComponentModel().GetService<IInterpreterRegistryService>();
                var interpreter = registry.FindConfiguration(runTarget.PythonInterpreter.Id);
                if (interpreter == null) {
                    return;
                }
                config = new LaunchConfiguration(interpreter);
            } else {
                config = new LaunchConfiguration(null);
            }

            config.InterpreterPath = runTarget.InterpreterPath;
            config.ScriptName = runTarget.Script;
            config.ScriptArguments = runTarget.Arguments;
            config.WorkingDirectory = runTarget.WorkingDirectory;

            RunProfiler(session, config, openReport);
```

So, if `runTarget.Interpreter` is null, it would create a blank config, and then set its `InterpreterPath` properly. But for some reason, that null branch is never taken anymore.

Reply:
Yeah that makes sense to me, I modified the code a little bit so it's working as expected now.